nservicebus/vscode.md

```diff
@@ -0,0 +1,263 @@
+* `cwd`: Specifies the current working directory for the launched `dotnet` process.
+* `stopAtEntry`: If set to true, the debugger will break at the entry point even without a breakpoint set.
+* `console`: Set to `externalTerminal` to use external console windows rather than VSCode's built-in terminal. This is useful when running multiple NServiceBus endpoints.
+
+### Web application
+
+For web application projects that need to launch a browser window the configuration is a little more involved:
+
+```json
+{
+    "name": "EShop.UI",
+    "type": "coreclr",
+    "request": "launch",
+    "preLaunchTask": "build",
+    "program": "${workspaceFolder}/path/to/Website/bin/Debug/netcoreapp2.1/Website.dll",
+    "args": [],
+    "cwd": "${workspaceFolder}/path/to/Website",
+    "stopAtEntry": false,
+    "internalConsoleOptions": "openOnSessionStart",
+    "launchBrowser": {
+        "enabled": true,
+        "args": "${auto-detect-url}",
+        "windows": {
+            "command": "cmd.exe",
+            "args": "/C start ${auto-detect-url}"
+        },
+        "osx": {
+            "command": "open"
+        },
+        "linux": {
+            "command": "xdg-open"
+        }
+    },
+    "env": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+    },
+    "sourceFileMap": {
+        "/Views": "${workspaceFolder}/path/to/Website/Views"
+    }
+}
+```
+
+* `internalConsoleOptions`: Assuming a single website project in a solution can use VSCode's integrated console, this option controls how it behaves. It's also possible to use the `console` property to use an external console.
+* `launchBrowser`: Contains instructions for launching a browser window on a platform-specific basis.
+* `env`: Additional environment variables to pass to the application, including the common `ASPNETCORE_ENVIRONMENT` variable that affects how an ASP.NET Core web application will read configuration values.
+* `sourceFileMap`: Additional source file mappings passed to the debugger, useful in this case to let the debugger know the path to MVC views.
+
+### Attaching to a process
+
+it's also possible to start debugging from a terminal and then attach to the process with this configuration:
+
+```json
+{
+    "name": ".NET Core Attach",
+    "type": "coreclr",
+    "request": "attach",
+    "processId": "${command:pickProcess}"
+}
+```
+
+### Compounds
+
+Running multiple NServiceBus endpoints together requires the `compounds` element, which is simply a `name` to appear in VSCode's Debug menu, and a list of configuration names specified elsewhere in the file:
+
+```json
+"compounds": [
+    {
+        "name": "Debug All",
+        "configurations": [
+            "Sales",
+            "Billing",
+            "Website"
+        ]
+    }
+]
+```
+
+### Complete example
+
+This snippet shows an entire launch.json file with **Sales** and **Billing** endpoints plus a **Website** project all in a directory named `src` beneath the project root:
```

Comment:
Do we need the word "snippet" here?

Reply:
Since you've been making "tasks.json" bold, seems like "launch.json" should be getting the same treatment throughout the doc as well.

Reply:
As the sentence is currently structured, I'm pretty sure it needs a comma somewhere. I'm thinking it's before "all" but I'm 100% sure on that.